.changeset/green-boats-attend.md

```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog-backend': patch
+---
+
+Support supplying a custom catalog descriptor file parser
```

Comment:
Could we have some documentation on how to pass in a custom parser?

Reply:
Hm not sure. It's a very fringe use case and there's no docs for the entire builder part at all - oh gosh I just now see that the installation part is way out of date overall :/ https://github.com/backstage/backstage/blob/master/docs/features/software-catalog/installation.md

Reply:
Let's have this as a todo to update the installation instructions, and until then, to rely on the doc comments for the builder

Reply:
A quick example would be really helpful. We have a format similar to `catalog-info.yaml` and so we don't want to duplicate the information. We've implemented a Processor that parses that format already, but other processors assume the `url` type (e.g. CodeOwners), and so we need to introduce hacks to integrate with them.

Using a Parser seems like a good alternative but I'm not sure how it's meant to be used. It seems the parser should be set via the CatalogBuilder, but then does that mean it's mutually exclusive with `catalog-info.yaml`? It would still be useful to support `catalog-info.yaml` for e.g. templates. Is it possible to have two catalogs that are merged together?